### PR TITLE
update testing to use upstream repository

### PR DESCRIPTION
**Pull request by mpougin**
update aiida-testing to track directly from https://github.com/aiidateam/aiida-testing as discussed with @ltalirz and  @chrisjsewell . Can you please double-check if the syntax is correct?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -41,7 +41,7 @@ Homepage = "https://github.com/lsmo-epfl/aiida-lsmo"
 [project.optional-dependencies]
 testing = [
     "pgtest~=1.3.1",
-    "aiida-testing @  git+https://github.com/ltalirz/aiida-testing@4212ba83896693ea13908eeb233ca3b35d504146", # egg=aiida-testing
+    "aiida-testing @  git+https://github.com/aiidateam/aiida-testing", # egg=aiida-testing
```
Comment (chrisjsewell) [inline pyproject.toml]: I think you currently need to use the develop branch, like
```suggestion
    "aiida-testing @  git+https://github.com/aiidateam/aiida-testing@develop", # egg=aiida-testing
```

Comment (mpougin) [reply pyproject.toml]: thanks @chrisjsewell

**Comment by mpougin** [comment]
are errors related to dependency conflicts?

**Comment by ltalirz** [comment]
The error seems to arise from this line
https://github.com/lsmo-epfl/aiida-lsmo/blob/47f080615fef31b144339a5dd2af64f3d6bb6ac6/conftest.py#L13
which is now no longer needed, since `aiida-testing` registers the fixtures automatically.

@chrisjsewell I am still not entirely convinced that saving the user from writing this one line is really worth the additional "magic" - and, as you can see, this can lead to confusion.
Should we revert it?

**Comment by mpougin** [comment]
> The error seems to arise from this line
> 
> https://github.com/lsmo-epfl/aiida-lsmo/blob/47f080615fef31b144339a5dd2af64f3d6bb6ac6/conftest.py#L13
> 
> which is now no longer needed, since `aiida-testing` registers the fixtures automatically.

Thanks for the clarification @ltalirz . From my point of you, as you said this is only confusing, and requires additional modification of the `conftest.py` file

**Comment by chrisjsewell** [comment]
> the additional "magic"

Its not magic, its exactly the way pytest says to make plugins: https://docs.pytest.org/en/7.1.x/how-to/writing_plugins.html#making-your-plugin-installable-by-others

it only leads to confusion, when it was done the wrong way intially

**Comment by mpougin** [comment]
> it only leads to confusion, when it was done the wrong way intially

Maybe it's just lack of experience but I wouldn't be able to tell from the error message/without knowing about the changes you made what the problem is

**Comment by ltalirz** [comment]
> > the additional "magic"
> 
> Its not magic, its exactly the way pytest says to make plugins: https://docs.pytest.org/en/7.1.x/how-to/writing_plugins.html#making-your-plugin-installable-by-others
> 
> it only leads to confusion, when it was done the wrong way intially

I see your point.

At the same time, I understand that with this approach there is no longer a way to turn off fixtures that ship with `autouse=True`, right?

If I want to run the test suite on any package I have installed in my Python environment (AiiDA-related or not), the `aiida_profile` fixture will now be executed before every test.

**Comment by chrisjsewell** [comment]
> the `aiida_profile` fixture will now be executed before every test.

yep exactly, the `aiida_profile` fixture should not be using `autouse=True` 😉  I'm not really sure why it was added

**Pull request closed without merging**